File: go/007.go
func reverse(x int) int {
    var ret int32
    for x != 0 {
        if abs(ret) > math.MaxInt32 / 10 {
            return 0
        }
        ret = ret * 10 + int32(x) % 10
        x /= 10
    }
    return int(ret)
}

func abs(x int32) int32 {
    if x < 0 {
        return -x
    }
    return x
}

File: cpp/007.cpp
class Solution {
public:
    int reverse(int x) {
        int ret = 0;
        while (x != 0) {
            if (abs(ret) > INT_MAX / 10 )
                return 0;
            ret = ret * 10 + x % 10;
            x /= 10;
        }
        return ret;
    }
};
